classes/controller/AdminController.php

```diff
@@ -4869,4 +4872,22 @@ private function prepareNumberSpecifications(Context $context)
             $numberSpecification->toArray()
         );
     }
+
+    /**
+     * Adds jQuery library file to queued JS file list.
+     *
+     * @param string|null $version jQuery library version
+     * @param string|null $folder jQuery file folder
+     * @param bool $minifier if set tot true, a minified version will be included
+     *
+     * @deprecated 1.7.7 jQuery is always included, this method should no longer be used
+     */
+    public function addJquery($version = null, $folder = null, $minifier = true)
```

Comment:
I understand it is deprecated, however it is only added when `setMedia` is called, but are we sure it is always called? What if a module controller only calls `addJquery` but not `setMedia`?
Maybe you should still add the required js files here:
```javascript
$this->addJs(_PS_JS_DIR_ . 'jquery/jquery-3.4.1.min.js');
$this->addJs(_PS_JS_DIR_ . 'jquery/jquery-migrate-3.1.0.min.js');
```

Since `addJs` checks the already added script it won't duplicate it any way.

Unless `setMedia` is always called by the controller class anyway, in which case forget my comment ;)

Reply:
@jolelievre I confirm that the setMedia method is always called form the controller in `ControllerCore::Run()`.